### ISSUE TITLE
How to retrieve results from adversarial training

### ISSUE THREAD

**Issue by jcpeterson** [revbucket/mister_ed#18]
They print out, but there's nothing returned by training_obj.train()

**Comment by revbucket** [comment]
I'm not sure I understand what exactly you want returned here. The classifier itself is mutated, and evaluation results need to be generated by evaluation explicitly, so I'm guessing you want the loss that's printed out during training. 

One simple way to do this is to redirect stdout before you call `.train(...)`, https://stackoverflow.com/questions/4675728/redirect-stdout-to-a-file-in-python, but I could also add some more logging features if you like

**Comment by jcpeterson** [comment]
I just mean if you want to plot both training loss/accuracy and validation loss/accuracy together. If the object is already printing loss/acc for each batch during training, it can return a list of scores after training for plotting.

**Comment by revbucket** [comment]
Sure, okay,  that's probably something good to have. I should have this out within the hour.

**Comment by jcpeterson** [comment]
Oh thanks. Not trying to rush you to do things btw :). I just like the package and am integrating into my current pipelines. 

Btw, on the same topic, for evaluation as you say, I imagine the model object is modified in place, such that I can call training_obj.train() with epochs set to 1, evaluate with an evaluation object, and then run another epoch?

**Comment by tianweiy** [comment]
We are glad that you like it. For the results of adversarial training, I personally want to include support for [visdom](https://github.com/facebookresearch/visdom) to help debugging and tuning parameters. We will first let the program output a list of scores and then add more visualization support like training visualization I mentioned above and [issue16.](https://github.com/revbucket/mister_ed/issues/16) This will take some time as I haven't used the visdom before. I don't know whether it is modified in place, let me do some experiment and see if it can and how to improve if it can't.

**Comment by tianweiy** [comment]
Yes, it is modified in place. You can also use the checkpoint in your pretrained model folder to do other experiments.

**Comment by revbucket** [comment]
Okay, done with some logging updates.

@jcpeterson : check out the most recent version of tutorial_2.ipynb to see how this works. Under the hood we just maintain a `TrainingLogger` instance from pytorch_utils.py, and the logging level is controlled the same way `[low, medium, high, snoop]` as print level during training.

**Comment by jcpeterson** [comment]
Wonderful thanks
